x-pack/plugins/index_management/public/application/components/mappings_editor/__jest__/client_integration/mappings_editor.test.tsx

```diff
@@ -166,12 +205,183 @@ describe('<MappingsEditor />', () => {
+        address: {
+          type: 'object',
+          properties: {
+            street: { type: 'text' },
+            city: { type: 'text' },
+          },
+        },
+      },
+      dynamic_templates: [{ initial: 'value' }],
+      _source: {
+        enabled: true,
+        includes: ['field1', 'field2'],
+        excludes: ['field3'],
+      },
+      _meta: {
+        some: 'metaData',
+      },
+      _routing: {
+        required: false,
+      },
+    };
+
+    let testBed: MappingsEditorTestBed;
+
+    beforeEach(async () => {
+      testBed = await setup({ value: defaultMappings, onChange: onChangeHandler });
+    });
+
+    test('props.value => should prepopulate the editor data', async () => {
+      const {
+        actions: { selectTab, getJsonEditorValue, getComboBoxValue, getToggleValue },
+        find,
+      } = testBed;
+
+      /**
+       * Mapped fields
+       */
+      expect(find('fieldsListItem').length).toEqual(Object.keys(defaultMappings.properties).length);
+
+      const fields = find('fieldsListItem.fieldName').map(item => item.text());
+      expect(fields).toEqual(Object.keys(defaultMappings.properties).sort());
+
+      /**
+       * Dynamic templates
+       */
+      await act(async () => {
+        await selectTab('templates');
+      });
+
+      const templatesValue = getJsonEditorValue('dynamicTemplatesEditor');
+      expect(templatesValue).toEqual(defaultMappings.dynamic_templates);
+
+      /**
+       * Advanced settings
+       */
+      await act(async () => {
+        await selectTab('advanced');
+      });
+
+      const isDynamicMappingsEnabled = getToggleValue(
+        'advancedConfiguration.dynamicMappingsToggle.input'
+      );
+      expect(isDynamicMappingsEnabled).toBe(defaultMappings.dynamic);
+
+      const isNumericDetectionEnabled = getToggleValue(
+        'advancedConfiguration.numericDetection.input'
+      );
+      expect(isNumericDetectionEnabled).toBe(defaultMappings.numeric_detection);
+
+      expect(getComboBoxValue('sourceField.includesField')).toEqual(
+        defaultMappings._source.includes
+      );
+      expect(getComboBoxValue('sourceField.excludesField')).toEqual(
+        defaultMappings._source.excludes
+      );
+
+      const metaFieldValue = getJsonEditorValue('advancedConfiguration.metaField');
+      expect(metaFieldValue).toEqual(defaultMappings._meta);
+
+      const isRoutingRequired = getToggleValue('advancedConfiguration.routingRequiredToggle.input');
+      expect(isRoutingRequired).toBe(defaultMappings._routing.required);
+    });
+
+    test('props.onChange() => should forward the changes to the consumer component', async () => {
+      let updatedMappings = { ...defaultMappings };
+
+      const {
+        actions: { addField, selectTab, updateJsonEditor },
+        component,
+        form,
+      } = testBed;
+
+      /**
+       * Mapped fields
+       */
+      const newField = { name: getRandomString(), type: 'text' };
```

Comment:
What do we get out of randomizing this value vs hardcoding it? When I was running through the CCR tests, it interfered with my pattern matching when comparing output to the objects defined in the tests as I tried to reason about failures, and I'm wondering if the benefits outweigh the costs.

Reply:
Why not? 😊 In this concrete case, I admit it does get in the way. I changed it to a hardcoded one.